docs/autogen_config.py

```diff
@@ -96,32 +98,39 @@ def reverse_aliases(app):
             if len(cls_cfg) == 1:
                 traitname = list(cls_cfg)[0]
                 if cls_cfg[traitname] is True:
-                    res[classname+'.'+traitname].append(flag)
+                    res[classname + "." + traitname].append(flag)
 
     return res
 
+
 def write_doc(name, title, app, preamble=None):
     trait_aliases = reverse_aliases(app)
-    filename = join(options, name+'.rst')
-    with open(filename, 'w') as f:
-        f.write(title + '\n')
-        f.write(('=' * len(title)) + '\n')
-        f.write('\n')
-        if preamble is not None:
-            f.write(preamble + '\n\n')
-        #f.write(app.document_config_options())
+    filename = options / (name + ".rst")
+
+    text = [title, "=" * len(title), "\n"]
+    if preamble is not None:
+        text.append(preamble + "\n")
+
+    text.append(app.document_config_options())
+
+    text.extend(
+        class_config_rst_doc(c, trait_aliases) for c in app._classes_inc_parents()
+    )
 
-        for c in app._classes_inc_parents():
-            f.write(class_config_rst_doc(c, trait_aliases))
-            f.write('\n')
+    filename.write_text("\n".join(text))
```

Comment:
I'm not quite sure all those are correct, as `write` likely does not happend \n, but '\n'.join() does, so this will not be strictly identical.

Reply:
Alright, can you revert back that change?

Reply:
yes, I can do that.